File: Blocks/CAB/VB/Source/CompositeUI.WinForms/Workspaces/ZoneSmartPartInfo.vb
'===============================================================================
' Microsoft patterns & practices
' CompositeUI Application Block
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================


Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.Windows.Forms
Imports Microsoft.Practices.CompositeUI.SmartParts
Imports System.Drawing


''' <summary>
''' Provides infomation to show smartparts in the <see cref="ZoneWorkspace"/>.
''' </summary>
<ToolboxItem(True), ToolboxBitmap(GetType(ZoneSmartPartInfo), "ZoneSmartPartInfo")> _
Public Class ZoneSmartPartInfo : Inherits SmartPartInfo
	Private innerZoneName As String
	Private innerDock As System.Nullable(Of DockStyle) = Nothing

	''' <summary>
	''' Initializes the ZonedSmartPartInfo with no zone name.
	''' </summary>
	Public Sub New()
	End Sub

	''' <summary>
	''' Creates the information for the smart part specifying the name of the zone.
	''' </summary>
	''' <param name="zoneName">Name of the zone assigned to a smart part.</param>
	Public Sub New(ByVal zoneName As String)
		Me.innerZoneName = zoneName
	End Sub

	''' <summary>
	''' Creates the information for the smart part specifying its title and the name of the zone.
	''' </summary>
	''' <param name="zoneName">Name of the zone assigned to a smart part.</param>
	''' <param name="title">Title of the smart part.</param>
	Public Sub New(ByVal title As String, ByVal zoneName As String)
		MyBase.Title = title
		Me.innerZoneName = zoneName
	End Sub

	''' <summary>
	''' Name of the zone where the smart part should be shown.
	''' </summary>
	''' <remarks>
	''' If a zone with the given name does not exist in the <see cref="ZoneWorkspace"/> 
	''' where the smart part is being shown, an exception will be thrown.
	''' </remarks>
	<Category("Layout"), DefaultValue(CType(Nothing, String))> _
	Public Property ZoneName() As String
		Get
			Return innerZoneName
		End Get
		Set(ByVal value As String)
			innerZoneName = value
		End Set
	End Property

	''' <summary>
	''' Sets the dockstyle of the control to show in the zone.
	''' </summary>
	<DefaultValue(0), Category("Layout")> _
	Public Property Dock() As System.Nullable(Of DockStyle)
		Get
			Return innerDock
		End Get
		Set(ByVal value As System.Nullable(Of DockStyle))
			innerDock = value
		End Set
	End Property
End Class

File: Blocks/CAB-WPFExtensions/Source/CompositeUI.WPF/Workspaces/ZoneSmartPartInfo.cs
//===============================================================================
// Microsoft patterns & practices
// Smart Client Software Factory 2010
//===============================================================================
// Copyright (c) Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
//===============================================================================
using System.ComponentModel;
using System.Windows.Forms;
using System.Drawing;

namespace Microsoft.Practices.CompositeUI.WPF
{
	/// <summary>
	/// Provides infomation to show smartparts in the <see cref="ZoneWorkspace"/>.
	/// </summary>
	public class ZoneSmartPartInfo : WPFSmartPartInfo
	{
		private string zoneName;
		private DockStyle? dock = null;

		/// <summary>
		/// Initializes the ZoneSmartPartInfo with no zone name.
		/// </summary>
		public ZoneSmartPartInfo()
		{
		}

		/// <summary>
		/// Creates the information for the smart part specifying the name of the zone.
		/// </summary>
		/// <param name="zoneName">Name of the zone assigned to a smart part.</param>
		public ZoneSmartPartInfo(string zoneName)
		{
			this.zoneName = zoneName;
		}

		/// <summary>
		/// Creates the information for the smart part specifying its title and the name of the zone.
		/// </summary>
		/// <param name="zoneName">Name of the zone assigned to a smart part.</param>
		/// <param name="title">Title of the smart part.</param>
		public ZoneSmartPartInfo(string title, string zoneName)
		{
			base.Title = title;
			this.zoneName = zoneName;
		}

		/// <summary>
		/// Name of the zone where the smart part should be shown.
		/// </summary>
		/// <remarks>
		/// If a zone with the given name does not exist in the <see cref="ZoneWorkspace"/> 
		/// where the smart part is being shown, an exception will be thrown.
		/// </remarks>
		public string ZoneName
		{
			get { return zoneName; }
			set { zoneName = value; }
		}

		/// <summary>
		/// Sets the dockstyle of the control to show in the zone.
		/// </summary>
		public DockStyle? Dock
		{
			get { return dock; }
			set { dock = value; }
		}
	}
}